packages/react/src/components/CodeSnippet/CodeSnippet.mdx

```diff
@@ -0,0 +1,41 @@
+import { Story, Props, Source, Preview } from '@storybook/addon-docs/blocks'
+import { withKnobs, boolean, select, text } from '@storybook/addon-knobs'
+import * as stories from './CodeSnippet-story.js'
+import CodeSnippet from '../CodeSnippet'
+
+<!-- How do I get <Props> to work?
+ -->
+# CodeSnippet
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Skeleton state](#skeleton-state)
+- [Component API](#component-api)
+- [Feedback](#feedback)
+
+## Overview
+Code snippets are strings or small blocks of reusable code that can be copied and inserted in a code file.
+
+<Preview>
+	<Story id="codesnippet--code-snippet" />
+</Preview>
+
```

Comment:
It'd be helpful to talk about the different variations as top-level headings, maybe the following each with their own default example:

- Inline
- Multi-line
- Single-line

At a top-level, it might be helpful to address a couple of common situations:

- Copying a code snippet
  - How to enable/disable
  - How to configure the feedback provided
- Rendering a "light" variant, and when/why to use that

Reply:
I added individual sections for the different use cases (inline, multi, light etc) -- seemed a little redundant to describe turning the copy button off and on and stuff with the API and expected boolean listed in the props table though, but I can go back and add that if we think it's something we need.